lib/actions/napster.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+function getMetadata(id, parentUri, type, title, cdudn) {
+  return `<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/"
+  xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/">
+  <item id="${id}" parentID="${parentUri}" restricted="true"><dc:title>"${title}"</dc:title><upnp:class>${type}</upnp:class>
+  <desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">${cdudn}</desc></item></DIDL-Lite>`;
+}
+
+function getSongUri(id) {
+  return `x-sonos-http:song%3a${id}.mp4?sid=204&flags=8224&sn=5`;
+}
+
+function getAlbumUri(id) {
+  return `x-rincon-cpcontainer:100420ecexplore%3aalbum%3a%3aAlb.${id}`;
+}
+
+
+
+const uriTemplates = {
+  song: getSongUri,
+  album: getAlbumUri
+};
+
+const CLASSES = {
+  song: 'object.item.audioItem.musicTrack',
+  album: 'object.container.album.musicAlbum'
+};
+
+const METADATA_URI_STARTERS = {
+  song: '10032020song%3a',
+  album: '100420ec'
+};
+
+const PARENTS = {
+  song: '1004206calbum%3a',
+  album: '100420ecexplore%3aalbum%3a%3a'
+};
+
+const CDUDN = {
+  song: 'SA_RINCON52231_X_#Svc52231-0-Token',
+  album: 'SA_RINCON55303_X_#Svc55303-0-Token'
```

Comment:
Also confused why these have two different values. SA_RINCONxxxxx should be the "type" of the service, and is 51975 on mine (they align with the ID), so 52231 might relate to ID 204, but not sure what 55303 maps to, should be a different ID (or sid, from the url).